src/ui/options/components/permissions.tsx

```diff
@@ -0,0 +1,37 @@
+import { Setter, Show, createSignal } from 'solid-js';
+import browser from 'webextension-polyfill';
+import styles from './components.module.scss';
+import { t } from '@/util/i18n';
+
+export default function Permissions() {
+	const [perms, setPerms] = createSignal(true);
+	hasPermissions(setPerms);
+
+	return (
+		<Show when={!perms()}>
```

Comment:
How does this work? how does `perms()` know what perms are set?

Reply:
perms is the getter function of a [signal](https://www.solidjs.com/tutorial/introduction_signals)

The signal is set to true by default, then the signal setter setPerms is uses to set it to false if perms are not granted, and set back to true if the user grants permission.

Signals allow the component to be reactive – solid only renders the component once, so if you just use a raw variable it will not change what is rendered when you change the variable value. Signals (and other similar solid utilities) tell the component to actually change what is rendered too.

Reply:
Thanks for the explanation thought it was something like that :)